consensus/ibft/src/integration-test/java/tech/pegasys/pantheon/consensus/ibft/tests/LocalNodeIsProposerTest.java

```diff
@@ -0,0 +1,116 @@
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.consensus.ibft.tests;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static tech.pegasys.pantheon.consensus.ibft.support.MessageReceptionHelpers.assertPeersReceivedExactly;
+import static tech.pegasys.pantheon.consensus.ibft.support.MessageReceptionHelpers.assertPeersReceivedNoMessages;
+
+import tech.pegasys.pantheon.consensus.ibft.ConsensusRoundIdentifier;
+import tech.pegasys.pantheon.consensus.ibft.IbftBlockHashing;
+import tech.pegasys.pantheon.consensus.ibft.IbftExtraData;
+import tech.pegasys.pantheon.consensus.ibft.ibftevent.BlockTimerExpiry;
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.CommitPayload;
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.MessageFactory;
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.ProposalPayload;
+import tech.pegasys.pantheon.consensus.ibft.ibftmessagedata.SignedData;
+import tech.pegasys.pantheon.consensus.ibft.support.RoundSpecificNodeRoles;
+import tech.pegasys.pantheon.consensus.ibft.support.TestContext;
+import tech.pegasys.pantheon.consensus.ibft.support.TestContextFactory;
+import tech.pegasys.pantheon.crypto.SECP256K1;
+import tech.pegasys.pantheon.crypto.SECP256K1.Signature;
+import tech.pegasys.pantheon.ethereum.core.Block;
+
+import java.time.Clock;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class LocalNodeIsProposerTest {
+  // These tests assume the basic function of the Ibft Round State Machine has been proven
+  // via the LocalNodeIsNotProposerTest.
+
+  private final long blockTimeStamp = 100;
+  private final Clock mockBlockCreationTime = mock(Clock.class);
+
+  // Local node will propose the first block
+  private final TestContext context =
+      TestContextFactory.createTestEnvironment(4, 1, mockBlockCreationTime);
+  private final ConsensusRoundIdentifier roundId = new ConsensusRoundIdentifier(1, 0);
+  private final RoundSpecificNodeRoles roles = context.getRoundSpecificRoles(roundId);
+
+  private final MessageFactory localNodeMessageFactory = context.getLocalNodeMessageFactory();
+
+  private Block expectedProposedBlock;
+  private SignedData<ProposalPayload> expectedTxProposal;
+  private SignedData<CommitPayload> expectedTxCommit;
+
+  @Before
+  public void setup() {
+    when(mockBlockCreationTime.millis()).thenReturn(blockTimeStamp * 1000);
```

Comment:
`blockTimeStamp * 1000` would be the 100 seconds ...what's the significance of that number?

Reply:
So - as the local node is the proposer, it will create a block, with the timestamp as reported by the Clock (which is mocked in these tests).

In order to validate packets, the timestamp in the created (and transmitted) block must known - thus the _expectedBlock_ has 'blockTimestamp' in it - and the clock which will provide a timestamp to the UUT is set to the same time.

Reply:
Could use Clock.fixed to created a clock for testing

Reply:
Right, so it's just about covering the `blocTimeStamp` to seconds ...because in that case you could use`TimeUnit.MILLISECONDS.toSeconds(blocTimeStamp)` and loose the magic number :)

Reply:
Changed to clock.fixed, nicer than using Mocks if we don't need to.